pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java

```diff
@@ -128,29 +129,37 @@ public QuotaCheckerResponse isSegmentStorageWithinQuota(@Nonnull File segmentFil
+        message = String.format("Refreshing segment %s of Table %s is within quota. Total allowed storage size: %s ( = configured quota: %s * number replicas: %d). New estimated table size of all replicas: %s. Current table size of all replicas: %s. Incoming uncompressed segment size of all replicas: %s ( = single incoming uncompressed segment size: %s * number replicas: %d). Existing same segment size of all replicas: %s. Formula: New estimated size = current table size - existing same segment size + incoming segment size",
+            segmentName, tableName, DataSize.fromBytes(allowedStorageBytes), DataSize.fromBytes(quotaConfig.storageSizeBytes()), numReplicas, DataSize.fromBytes(estimatedFinalSizeBytes), DataSize.fromBytes(tableSubtypeSize.estimatedSizeInBytes), DataSize.fromBytes(totalIncomingSegmentSizeBytes), DataSize.fromBytes(incomingSegmentSizeBytes), numReplicas, DataSize.fromBytes(existingSegmentSizeBytes));
+      }
+      LOGGER.warn(message);
       return new QuotaCheckerResponse(true, message);
     } else {
       String message;
       if (tableSubtypeSize.estimatedSizeInBytes > allowedStorageBytes) {
         message = String.format(
-            "Table %s already over quota. Existing estimated uncompressed size of all replicas: %d bytes > total allowed storage size: %d bytes ( = configured quota: %d bytes * num replicas: %d). Check if indexes were enabled recently and adjust table quota accordingly.",
-            tableName, tableSubtypeSize.estimatedSizeInBytes, allowedStorageBytes, quotaConfig.storageSizeBytes(),
+            "Table %s already over quota. Existing estimated uncompressed table size of all replicas: %s > total allowed storage size: %s ( = configured quota: %s * num replicas: %d). Check if indexes were enabled recently and adjust table quota accordingly.",
+            tableName, DataSize.fromBytes(tableSubtypeSize.estimatedSizeInBytes), DataSize.fromBytes(allowedStorageBytes), DataSize.fromBytes(quotaConfig.storageSizeBytes()),
             numReplicas);
       } else {
         message = String.format(
-            "Storage quota exceeded. Newly estimated size: %d bytes ( = existing estimated uncompressed size of all replicas: %d bytes - (existing segment sizes of all replicas: %d bytes) + (incoming compressed segment size: %d bytes * number replicas: %d)) > total allowed storage size: %d bytes ( = configured quota: %d bytes * number replicas: %d) for table %s.",
-            estimatedFinalSizeBytes, tableSubtypeSize.estimatedSizeInBytes, existingSegmentSizeBytes,
-            incomingSegmentSizeBytes, numReplicas, allowedStorageBytes, quotaConfig.storageSizeBytes(), numReplicas,
+            "Storage quota exceeded. Newly estimated size: %s ( = existing estimated uncompressed size of all replicas: %s - (existing segment sizes of all replicas: %s) + (incoming compressed segment size: %s * number replicas: %d)) > total allowed storage size: %s ( = configured quota: %s * number replicas: %d) for table %s.",
```

Comment:
Thanks for including an example in the description. I had one additional suggestion to make this more readable: "Newly estimated size: %s > total allowed %s. New estimated size = (<numbers that we used to arrive at it>), total allowed=(<numbers used to arrive at it).

This way, the main info is available upfront and the details are there for those who need it.